### PR TITLE
Game loop refactor

### PR DESCRIPTION
**Pull request by Mixss**
# Pull request review checklist 📝 ✅

This is a checklist that helps conduct a pull request review properly. These points are required to merge the PR

- [ ] There are no merge conflicts

- [ ] The pull request is named accordingly (description of the functionality - it
   will be visible in the changelog later)
   
- [ ] Every task requirement (from Jira) is done

- [ ] The app runs, it boots up properly (e.g. no syntax errors)

- [ ] The proposed functionality is working properly (you need to pull this branch
   and check for yourself)

- [ ] Each function, class or functionality is properly documented
   (short description of what it does, parameters or return values if they need
   any explanation)

### CODE REVIEW

File: dnd-bot/dnd_bot/dc/ui/message_templates.py [hunk 1]
```diff
@@ -124,6 +122,16 @@ def skills_message_template(player):
                                description=desc)
         return embed
 
+    @staticmethod
+    def creature_turn_embed(player: Player, active_creature: Creature, active_user_icon=None, recent_action=''):
+        """message embed representing the active player actions and the player's stats"""
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/dc/ui/message_templates.py:127]:
```suggestion
        """message embed representing the active creature's actions and the player's stats"""
```

File: dnd-bot/dnd_bot/dc/ui/message_templates.py [hunk 2]
```diff
@@ -124,6 +122,16 @@ def skills_message_template(player):
...
+        embed = nextcord.Embed(title=f'Position: ({player.x}, {player.y}) | Action points: {player.action_points}/'
+                                     f'{player.initial_action_points} | '
+                                     f'HP: {player.hp}/{player.hp}', description=recent_action)
+        embed.set_footer(text=f'{active_creature.name}\'s turn', icon_url=active_user_icon)
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/dc/ui/message_templates.py:131]: raczej bez ikonki, może generować błąd dla creatures

Comment (Mixss) [reply dnd-bot/dnd_bot/dc/ui/message_templates.py:131]: parametr active_user_icon jest domyślnie ustawiony na None, jest ustawiany jedynie kiedy typem active_creature jest Player

File: dnd-bot/dnd_bot/dc/ui/views/view_lobby.py [hunk 3]
```diff
@@ -92,40 +91,15 @@ def __init__(self, token):
         super().__init__()
         self.value = None
         self.token = token
+        self.auto_defer = True
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/dc/ui/views/view_lobby.py:94]: To jest cały czas wymagane?

Comment (Mixss) [reply dnd-bot/dnd_bot/dc/ui/views/view_lobby.py]: Nie

File: dnd-bot/dnd_bot/logic/prototype/game.py [hunk 4]
```diff
@@ -97,9 +98,6 @@ def get_movable_entities(self):
                     movable_entities.append(entity)
         return movable_entities
 
-    def get_active_player(self):
+    def get_active_creature(self):
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/prototype/game.py:101]: ta funkcja jest useless, pole active_creature nie jest prywatne

Comment (Mixss) [reply dnd-bot/dnd_bot/logic/prototype/game.py:101]: Nie jest, funkcja używana jest w `view_game.py`

File: dnd-bot/dnd_bot/logic/game/handler_game.py [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
+
+        if len(game.creatures_queue) == 0:
+            GameLoop.prepare_queue(game)
+
+        next_creature = game.creatures_queue.popleft()
+
+        recent_action_message = MessageTemplates.end_turn_recent_action_message(game.active_creature)
+        await ViewMain.display_views_for_users(game_token, next_creature, recent_action_message)
+
+        # reset creature's stats
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/game/handler_game.py:22]:
```suggestion
        # reset creature's action points to the initial value
```

**Review by Silentsky0 — APPROVED** [review]

**Pull request merged**